pkg/controllers/gacha-controller.go

```diff
@@ -110,7 +103,12 @@ func ListCharacters(w http.ResponseWriter, r *http.Request) {
 }
 
 func HandleGachaDraw(w http.ResponseWriter, r *http.Request) {
-
+	user, ok := r.Context().Value("user").(models.User)
+	if !ok {
+		http.Error(w, "User not found", http.StatusUnauthorized)
```

Comment:
What would be the situation without users?

Reply:
@GuiltyMorishita I think we would always require a user to be authenticated before drawing a gacha. So if there are no users then nobody can draw a gacha.